pkg/lib/hooks.js

```diff
@@ -324,3 +324,36 @@ export function useEvent(obj, event, handler) {
 export function useInit(func, deps, comps, destroy = null) {
     return useObject(func, destroy, deps || [], comps);
 }
+
+/* - usePolkitPermissions(actions)
+*
+ * function Component(arg) {
+ *   const permissions = usePolkitPermissions([
+ *     'org.freedesktop.NetworkManager.reload',
+ *     'org.freedesktop.NetworkManager.network-control'
+ *   ]);
+ *
+ *   ...
+ *  }
+ *
+ * The returned value from the hook is an array of booleans,
+ * in the order that they were specified, specifying if the
+ * action is allowed or not.
+ * ]);
+*/
+export function usePolkitPermissions(actions) {
+    const [permissions, setPermissions] = useState([]);
+
+    useEffect(() => {
+        Promise.allSettled(actions.map((action) =>
+            // pkcheck returns a 0 status code if the check passed
+            cockpit.spawn(['sh', '-c', `pkcheck --action-id ${action} --process $$ --allow-user-interaction 2>&1`])
```

Comment:
pkg/networkmanager/firewall-client.js calls that with `superuser: "try"`, so that it automatically "falls back" to sudo or having a root bridge in general. Depending on how you want to call this, I imagine it makes the callers easier.

Reply:
On second thought, no -- *if* we have superuser already, then there's no point in doing n pkcheck calls, it would just be a waste. So maybe the effect should check `superuser.alllowed` itself? :thinking:

Reply:
I did toy with the idea of doing `superuser: "try"`, however if you're in limited access mode but you have sudo permissions and not polkit, this would show as being allowed to the user.

If we did that, what would be the side effect of the calls to NetworkManager via dbus? Given the `cockpit-session` is run as the user account, would dbus fail in a weird way?

Reply:
I do like the idea of checking for `superuser.allowed` first though and just returning early

Reply:
The actual D-Bus calls always happen via the same connection object, which is opportunistically using root:
```
firewalld_dbus = cockpit.dbus('org.fedoraproject.FirewallD1', { superuser: "try" });
```

I.e. if you have root, then your D-Bus connection (and hence the calls) happen as root. If not, then firewalld (or NM in your case) will ask polkit whether the caller has permissions. But that happens on the server side.

The pkchecks -- at least the ones currently on the firewalld page -- are *only* influencing the presentation, i.e. deciding whether or not to show actions. In other words, they try to predict whether the D-Bus call as user will fail or not, so that we avoid showing actions in the UI which will fail.

Reply:
Agreed on the firewalld page, but is that the case for NetworkManager? I can't easily trace where that dbus instantiation is?

Reply:
It's here: https://github.com/cockpit-project/cockpit/blob/main/pkg/networkmanager/interfaces.js#L166

Same approach really, it's a common pattern.